source/beamline/AMSlits.cpp

```diff
@@ -181,16 +181,20 @@ AMControl* AMSlits::horizontalCenter() const
 	return horizontalSlit_->center();
 }
 
-void AMSlits::addChildControl(AMControl *control)
+bool AMSlits::addChildControl(AMControl *control)
 {
-	if (allControls_->addControl(control))
-		AMControl::addChildControl(control);
+	if (AMControl::addChildControl(control)){
+		return allControls_->addControl(control);
+	}
+	return false;
 }
 
-void AMSlits::removeChildControl(AMControl *control)
+bool AMSlits::removeChildControl(AMControl *control)
 {
-	if (!allControls_->removeControl(control))
-		AMControl::removeChildControl(control);
+	if (AMControl::removeChildControl(control)){
+		return !allControls_->removeControl(control);
```

Comment:
Looking at these closer I think they should be set back to
`if(!allControls_->removeControl(control)){
     return AMControl::removeChildControl(control);`
Otherwise 'allControls_->remoteControl(control)' tries to evaluate a non-existent control.